hardware/simulation/sim_setup.py

```diff
@@ -0,0 +1,24 @@
+# This script is called during setup.
+# You can use 'setup_module' to access the contents of the iob_soc_setup.py python module
+import os
+import shutil
+
+# Delete all test*.expected files from build_dir
+dirpath=os.path.join(setup_module.build_dir, "hardware/simulation/src")
+for file in os.listdir(dirpath):
+    if file.startswith("test") and file.endswith(".expected"):
+        os.remove(os.path.join(dirpath,file))
+
+# Find out correct test.expected filename
+test_file_name='test'
+#Check if setup with INIT_MEM (check if macro exists)
+if next((i for i in setup_module.confs if i['name']=='INIT_MEM'), False) and setup_module.confs['INIT_MEM']['val'] != 'NA':
```

Comment:
```suggestion
macro = next((i for i in setup_module.confs if i['name']=='INIT_MEM'), False)
if macro and macro['val'] != 'NA':
```

Reply:
I had updated these requested changes previously, but seems they were merged before I updated.
(The previous suggested change had the error `confs['INIT_MEM']`)